packages/react/src/components/Tag/Tag.js

```diff
@@ -50,40 +51,67 @@ const Tag = ({
       event.stopPropagation();
       onClose(event);
     }
-  };
-  return filter ? (
-    <div
-      className={tagClasses}
-      aria-label={
-        title !== undefined
-          ? `${title} ${children}`
-          : `Clear filter ${children}`
-      }
-      id={tagId}
-      {...other}>
+  }
+
+  if(filter) {
+    return (
+      <div
+        className={tagClasses}
+        aria-label={
+          title !== undefined
+            ? `${title} ${children}`
+            : `Clear filter ${children}`
+        }
+        id={tagId}
+        {...other}>
+        <span
+          className={`${prefix}--tag__label`}
+          title={typeof children === 'string' ? children : null}>
+          {children !== null && children !== undefined ? children : TYPES[type]}
+        </span>
+        <button
+          type="button"
+          className={`${prefix}--tag__close-icon`}
+          onClick={handleClose}
+          disabled={disabled}
+          aria-labelledby={tagId}
+          title={title}>
+          <Close16 />
+        </button>
+      </div>
+    );
+  } else if(CustomIconElement) {
```

Comment:
```suggestion
  }
  if (CustomIconElement) {
```

since we `return` in the block above, this following `else-if` block can just stand alone as an `if` block, if that makes sense